viaf.py

```diff
@@ -751,39 +783,42 @@ def lookup_by_viaf(self, viaf, working_sort_name=None,
         xml = r.content
         return self.parser.parse(xml, working_sort_name, working_display_name)
 
-
-    def lookup_by_name(self, sort_name, display_name=None, do_get=None):
-        name = sort_name or display_name
-        # from OCLC tech support:  
-        # VIAF's SRU endpoint can only return a maximum number of 10 records when the recordSchema is http://viaf.org/VIAFCluster
+    def lookup_by_name(self, sort_name, display_name=None, do_get=None,
+                       best_match=False):
+        sort_name = sort_name or display_name
+        # from OCLC tech support:
+        # VIAF's SRU endpoint can only return a maximum number of 10 records
+        # when the recordSchema is http://viaf.org/VIAFCluster
         maximum_records = 10 # viaf maximum that's not ignored
         page = 1
         contributor_candidates = []
 
-        # limit ourselves to reading the first 500 viaf clusters, on the assumption that 
-        # search match quality is unlikely to be usable after that.
-        #for page in range (20, 22):
+        # limit ourselves to reading the first 500 viaf clusters, on the
+        # assumption that search match quality is unlikely to be usable after that.
         for page in range (1, 51):
             start_record = 1 + maximum_records * (page-1)
             scope = 'local.personalNames'
             if is_corporate_name(sort_name):
                 scope = 'local.corporateNames'
 
-            url = self.SEARCH_URL.format(scope=scope, sort_name=name.encode("utf8"), maximum_records=maximum_records, start_record=start_record)
+            url = self.SEARCH_URL.format(
+                scope=scope, sort_name=sort_name.encode("utf8"),
+                maximum_records=maximum_records, start_record=start_record
+            )
             representation, cached = Representation.get(self._db, url, do_get=do_get)
             xml = representation.content
 
             candidates = self.parser.parse_multiple(xml, sort_name, display_name, page)
             if not any(candidates):
                 # Delete the representation so it's not cached.
-                self._db.query(Representation).filter(Representation.id==representation.id).delete()
-                # we ran out of clusters, so we can relax and move on to ordering the returned results
+                self._db.query(Representation).filter(
+                    Representation.id==representation.id
+                ).delete()
+                # We ran out of clusters, so we can relax and move on to
+                # ordering the returned results
                 break
 
             contributor_candidates.extend(candidates)
             page += 1
 
-        return contributor_candidates
-
-
-
+        return self.select_best_match(contributor_candidates, sort_name)
```

Comment:
I bet Darya's opinion on this will be a lot stronger than mine, but what's the advantage of calling select_best_match here (and changing the contract of the method) instead of calling it on line 201, the code that calls the method?